File: sdk/cpp/examples/plot/line.cpp
// OA_DOC_BEGIN: plot-line
#include <oa/runtime/engine.h>
#include <oa/ui/plot/plot.h>

#include <array>
#include <cstdio>
#include <utility>

int main() {
	oa::EngineConfig engineConfig;
	engineConfig.appName = "ExamplePlotLine";
	engineConfig.presentationMode = oa::PresentationMode::Headless;

	auto created = oa::Engine::create(engineConfig);
	if (not created.isOk()) return 1;
	auto engine = std::move(created).getValue();

	oa::plot::Figure figure({
		.title = "training loss",
		.width = 360,
		.height = 240,
		.theme = oa::plot::Theme::Dark,
	});
	constexpr std::array<oa::F32, 6> loss{1.0F, 0.72F, 0.51F, 0.38F, 0.29F, 0.23F};
	figure.ax(0, 0).xLabel("step");
	figure.ax(0, 0).yLabel("loss");
	figure.ax(0, 0).plot(loss);

	auto rendered = figure.render(*engine);
	if (not rendered.isOk()) return 1;
	auto image = std::move(rendered).getValue();
	if (not image.validate() || image.width() != 360 || image.height() != 240) {
		return 1;
	}

	std::puts("rendered a 360x240 training-loss figure");
	return 0;
}
// OA_DOC_END: plot-line

File: sdk/py/examples/plot/line.py
#!/usr/bin/env python3
"""Render one retained plot figure into a semantic image."""

# OA_DOC_BEGIN: plot-line
from oa import initComputeEngine, plot, shutdownComputeEngine


if not initComputeEngine():
	raise RuntimeError("OA could not create a Vulkan compute engine")

try:
	config = plot.FigureConfig()
	config.title = "Training loss"
	config.width = 360
	config.height = 240
	config.theme = plot.Theme.Dark

	figure = plot.Figure(config)
	figure.ax(0, 0).xLabel("step")
	figure.ax(0, 0).yLabel("loss")
	figure.ax(0, 0).plot([1.0, 0.72, 0.51, 0.38, 0.29, 0.23])

	image = figure.render()
	assert image.validate()
	assert image.width() == 360
	assert image.height() == 240
	print("rendered a 360x240 training-loss figure")
finally:
	shutdownComputeEngine()
# OA_DOC_END: plot-line
